cloudconfig/containerinit/container_userdata.go

```diff
@@ -241,24 +241,31 @@ func shutdownInitCommands(initSystem, series string) ([]string, error) {
 		// keep clean logs for diagnosing issues / debugging.
 		"/var/log/cloud-init*.log",
 	}
+	paths := strings.Join(cleanupOnShutdown, " ")
+	removeCmd := fmt.Sprintf("/bin/rm -fr %s\n  ", paths)
+	shutdownCmd := "/sbin/shutdown -h now"
+	name := "juju-template-restart"
+	desc := "juju shutdown job"
 
 	// Using EOC below as the template shutdown script is itself
 	// passed through cat > ... < EOF.
 	replaceNetConfCmd := fmt.Sprintf(
 		"/bin/cat > /etc/network/interfaces << EOC%sEOC\n  ",
 		defaultEtcNetworkInterfaces,
 	)
-	paths := strings.Join(cleanupOnShutdown, " ")
-	removeCmd := fmt.Sprintf("/bin/rm -fr %s\n  ", paths)
-	shutdownCmd := "/sbin/shutdown -h now"
-	name := "juju-template-restart"
-	desc := "juju shutdown job"
 
-	execStart := shutdownCmd
+	// This intentionally avoids a for loop, keeping the ExecStart validation code more generic.
+	ifaces := `$(/usr/bin/awk '/^iface.*dhcp$/ {print $2}' /etc/network/interfaces /etc/network/interfaces.d/*)`
+	dhcpRelease := fmt.Sprintf("/sbin/dhclient -r %s &&\n  /sbin/ifdown %s\n  ", ifaces, ifaces)
```

Comment:
Mostly to force the release and only touch the ifaces configured with dhcp in ENI. I'm not sure I always saw the release on ifdown in testing, hence the forced release first.

This is limited to Juju 1.25, but it's possible this will need a rework if we find changes in OS releases while juju-1 is still supported.